File: src/main/java/edu/kit/iti/formal/stvs/logic/specification/smtlib/OptionalConcreteSpecificationHandler.java
package edu.kit.iti.formal.stvs.logic.specification.smtlib;

import edu.kit.iti.formal.stvs.model.table.ConcreteSpecification;

import java.util.Optional;

/**
 * Created by leonk on 22.02.2017.
 */
@FunctionalInterface
public interface OptionalConcreteSpecificationHandler{
  /**
   * Performs this operation on the given concreteSpecification.
   *
   * @param optionalConcreteSpecification the concreteSpecification argument
   */
  void accept(Optional<ConcreteSpecification> optionalConcreteSpecification);
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/logic/specification/smtlib/OptionalConcreteSpecificationHandler.kt
package edu.kit.iti.formal.stvs.logic.specification.smtlib;

import edu.kit.iti.formal.stvs.model.table.ConcreteSpecification;

import java.util.Optional;

/**
 * Is notified by the {@link Z3Solver} after concretization.
 */
@FunctionalInterface
public interface OptionalConcreteSpecificationHandler {
  /**
   * Performs this operation on the given concreteSpecification.
   *
   * @param optionalConcreteSpecification the concreteSpecification argument
   */
  void accept(Optional<ConcreteSpecification> optionalConcreteSpecification);
}
